plugins/javaembed/javaembed.cpp

```diff
@@ -1694,14 +1994,404 @@ class JavaThreadContext
     jmethodID javaMethodID;
 };
 
+class JavaXmlBuilder : public CInterface, implements IXmlWriterExt
+{
+public:
+    IMPLEMENT_IINTERFACE;
+    JavaXmlBuilder(JNIEnv *_JNIenv, IEsdlDefinition *esdl_, const char *esdlservice, const char *esdltype_)
+    : JNIenv(_JNIenv), esdl(esdl_), javaPackage(esdlservice), esdlType(esdltype_)
+    {
+    }
+    ~JavaXmlBuilder()
+    {
+        while (defStack.length())
+            popDefStackEntry(JNIenv);
+        HashIterator it(javaClasses);
+        ForEach(it)
+        {
+            IMapping &entry = it.query();
+            jclass *pClass = javaClasses.mapToValue(&entry);
+            if (pClass)
+                JNIenv->DeleteGlobalRef(*pClass);
+        }
+    }
+    void checkException()
+    {
+        javaembed::checkException(JNIenv, false);
+    }
+
+    void initWriter()
+    {
+    }
+    IXmlWriterExt & clear()
+    {
+        UNIMPLEMENTED;
```

Comment:
Does this need to be implemented sometime, or should it be a throwUnexpected(); ?